### PR TITLE
Added doc and examples for tf.keras.losses.get

### LINKED ISSUE #37240: How to deserialize from a dict with tf.keras.losses.get
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/losses/get

## Description of issue (what needs changing):
Currently there is no documentation at all.
Its fairly straightforward to use by inputting a string denoting default class name:
ex:
```
identifier = "categorical_crossentropy"
tf.keras.losses.get(identifier)
```

However, I am having issues with dictionary objects:
ex:
```
identifier = {"class_name":"categorical_crossentropy","config":{"from_logits":True}}
tf.keras.losses.get(identifier)
```
Returns:
```
Traceback (most recent call last):
  File ".\main.py", line 85, in <module>
    loss = tf.keras.losses.get(jsn)
  File "C:\Users\jopatterson\Documents\autoprime-ml\env\lib\site-packages\tensorflow_core\python\keras\losses.py", line 1186, in get
    return deserialize(identifier)
  File "C:\Users\jopatterson\Documents\autoprime-ml\env\lib\site-packages\tensorflow_core\python\keras\losses.py", line 1175, in deserialize
    printable_module_name='loss function')
  File "C:\Users\jopatterson\Documents\autoprime-ml\env\lib\site-packages\tensorflow_core\python\keras\utils\generic_utils.py", line 315, in deserialize_keras_object
    return cls(**cls_config)
TypeError: categorical_crossentropy() missing 2 required positional arguments: 'y_true' and 'y_pred'
```
I believe it is failing because cls is the already initialized loss function, and it is passing cls_config as its input, rather than using them as parameters during initialization.

### Clear description

This is a very useful method for abstract implementations of loss objects.

### Correct links

This is where the issue is occuring, within the `deserialize_keras_object` function:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/utils/generic_utils.py#L382

### Parameters defined

There currently is no documentation for this, as `identifier` can be a string, dictionary or callable.

### Returns defined

Returns are not defined, but its fairly obvious it returns a loss function.

### Raises listed and defined

No.

### Usage example

No.

### Request visuals, if applicable

No.

### Submit a pull request?

I would do this if I had enough knowledge to do so. Unfortunately I only know how it works with `identifier` as a String.

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #37240. 
@k-w-w , @mihaimaruseac - Please review this one.

### CODE REVIEW

File: tensorflow/python/keras/losses.py [hunk 1]
```diff
@@ -1843,15 +1843,35 @@ def deserialize(name, custom_objects=None):
 
 @keras_export('keras.losses.get')
 def get(identifier):
-  """Retrieves a Keras loss function.
+  """Retrieves a Keras loss as a `function`/`Loss` class instance.
+
+  You can get loss as a `function` using parameter `identifier` as a string
+  of the loss function as shown in below example.
+  >>> loss = tf.keras.losses.get("categorical_crossentropy")
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/losses.py]: Please add an empty line before every doctest to not break formatting on the website

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by ashutosh1919** [comment]
@mihaimaruseac , Please approve it again. Added empty lines wherever needed.

File: tensorflow/python/keras/losses.py [hunk 2]
```diff
@@ -1843,15 +1843,35 @@ def deserialize(name, custom_objects=None):
...
+  >>> type(loss)
+  function
+
+  You can get loss as a `Loss` class instance using parameter `identifier`
+  as a string of the class name of the loss.
+  >>> loss = tf.keras.losses.get("CategoricalCrossentropy")
+  >>> type(loss)
+  tensorflow.python.keras.losses.CategoricalCrossentropy
```
Comment (k-w-w) [inline tensorflow/python/keras/losses.py]: nit: I'd combine these examples e.g. 
```
The identifier may be the string name of a loss function or `Loss` class.

>>...
```

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/python/keras/losses.py [hunk 3]
```diff
@@ -1843,15 +1843,35 @@ def deserialize(name, custom_objects=None):
...
+
+  You can also specify `config` of the loss to this function by passing dict
+  containing `class_name` and `config` as an identifier.
```
Comment (k-w-w) [inline tensorflow/python/keras/losses.py]: Also note that the class name must map to a `Loss` class

**Review by k-w-w — COMMENTED** [review]
Thanks for adding this PR! A couple nits

File: tensorflow/python/keras/losses.py (hunk 3 above)
Comment (ashutosh1919) [reply tensorflow/python/keras/losses.py]: Done

File: tensorflow/python/keras/losses.py (hunk 2 above)
Comment (ashutosh1919) [reply tensorflow/python/keras/losses.py]: Done

**Review by mihaimaruseac — DISMISSED** [review]

**Review by k-w-w — DISMISSED** [review]

**Comment by ashutosh1919** [comment]
@mihaimaruseac and @k-w-w , Very sorry. Resolved doctest

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/keras/losses.py [hunk 4]
```diff
@@ -1843,15 +1843,34 @@ def deserialize(name, custom_objects=None):
...
+  """Retrieves a Keras loss as a `function`/`Loss` class instance.
+
+  The `identifier` may be the string name of a loss function or `Loss` class.
+
+  >>> loss = tf.keras.losses.get("categorical_crossentropy")
+  >>> type(loss)
+  <class 'function'>
+  >>> loss = tf.keras.losses.get("CategoricalCrossentropy")
+  >>> type(loss)
+  <class 'tensorflow.python.keras.losses.CategoricalCrossentropy'>
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/losses.py]: Inside Google the class is different (has a prefix). Let's replace with `<class '...tensorflow.python.keras.losses.CategoricalCrossentropy'>`

(hopefully this works, if it doesn't we'll have to rewrite the test somehow

File: tensorflow/python/keras/losses.py [hunk 5]
```diff
@@ -1843,15 +1843,34 @@ def deserialize(name, custom_objects=None):
...
+
+  You can also specify `config` of the loss to this function by passing dict
+  containing `class_name` and `config` as an identifier. Also note that the
+  `class_name` must map to a `Loss` class
+
+  >>> identifier = {"class_name": "CategoricalCrossentropy",
+  ...               "config": {"from_logits": True}}
+  >>> loss = tf.keras.losses.get(identifier)
+  >>> type(loss)
+  <class 'tensorflow.python.keras.losses.CategoricalCrossentropy'>
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/losses.py]: Also here

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by ashutosh1919** [comment]
@mihaimaruseac , Changed class type as per your comments. Please run ci. Let's see if it works, if it's  not, then I will think about better way to write these examples.

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**